node/src/components/consensus/highway_core/state/tests.rs

```diff
@@ -845,6 +845,29 @@ fn test_log2() {
     assert_eq!(62, log2((1 << 63) - 1));
 }
 
+#[test]
+fn test_leader() {
+    let weights = &[Weight(3), Weight(4), Weight(5)];
+
+    // All three validators get slots in the leader sequence.
+    let state = State::<TestContext>::new(weights, test_params(0), vec![], vec![]);
+    assert_eq!(
+        vec![CAROL, CAROL, CAROL, CAROL, CAROL, ALICE, BOB, ALICE, CAROL, ALICE],
+        (0..10u64)
+            .map(|r_id| state.leader(r_id.into()))
+            .collect_vec()
+    );
+
+    // If Carol is excluded, her slots get reassigned, but Alice and Bob keep their old slots.
+    let state = State::<TestContext>::new(weights, test_params(0), vec![], vec![CAROL]);
+    assert_eq!(
+        vec![ALICE, BOB, BOB, BOB, ALICE, ALICE, BOB, ALICE, ALICE, ALICE],
+        (0..10u64)
+            .map(|r_id| state.leader(r_id.into()))
+            .collect_vec()
+    );
+}
```

Comment:
Can you add a test where excluded validators are from the middle of the sequence as well? And with more than one? Something like `[V1, V2_excluded, V3_excluded, V4, V5_excluded]`?